services/dis-apim-operator/internal/controller/apiversion_controller.go

```diff
@@ -61,3 +107,185 @@ func (r *ApiVersionReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Named("apiversion").
 		Complete(r)
 }
+
+func (r *ApiVersionReconciler) deleteApiVersion(ctx context.Context, apiVersion apimv1alpha1.ApiVersion) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("Deleting APIVersion")
+	_, err := r.apimClient.DeleteApi(ctx, apiVersion.GetApiVersionAzureFullName(), "*", nil)
+	if azure.IgnoreNotFound(err) != nil {
+		logger.Error(err, "Failed to delete APIVersion")
+		return ctrl.Result{}, err
+	}
+	if apiVersion.Spec.Policies != nil {
+		_, err = r.apimClient.DeleteApiPolicy(ctx, apiVersion.GetApiVersionAzureFullName(), "*", nil)
+		if azure.IgnoreNotFound(err) != nil {
+			logger.Error(err, "Failed to delete policy")
+			return ctrl.Result{}, err
+		}
+	}
+	controllerutil.RemoveFinalizer(&apiVersion, API_VERSION_FINALIZER)
+	err = r.Update(ctx, &apiVersion)
+	if err != nil {
+		logger.Error(err, "Failed to remove finalizer")
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{}, nil
+}
+
+func (r *ApiVersionReconciler) handleApiVersionUpdate(ctx context.Context, apiVersion apimv1alpha1.ApiVersion) (ctrl.Result, error) {
+	latestSha, err := utils.Sha256FromContent(ctx, apiVersion.Spec.Content)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to get api spec sha: %w", err)
+	}
+	if apiVersion.Status.LastAppliedSpecSha != latestSha {
+		return r.createUpdateApimApi(ctx, apiVersion)
+	}
+	if apiVersion.Spec.Policies != nil {
+		_, policyErr := r.apimClient.GetApiPolicy(ctx, apiVersion.GetApiVersionAzureFullName(), nil)
+		lastPolicySha, shaErr := utils.Sha256FromContent(ctx, apiVersion.Spec.Policies.PolicyContent)
+		if shaErr != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to get policy sha: %w", shaErr)
+		}
+		if apiVersion.Spec.Policies != nil && apiVersion.Status.LastAppliedPolicySha != lastPolicySha || azure.IsNotFoundError(policyErr) {
```

Comment:
why not this?
```suggestion
		if apiVersion.Status.LastAppliedPolicySha != lastPolicySha || azure.IsNotFoundError(policyErr) {
```
Unless there's something I'm missing